testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -289,6 +289,44 @@
        and contains(@class,'node')]/ul/li/div/a[contains(.,'System List')]").click
 end
 
+#
+# Menu links
+#
+
+# Menu link: 1st level
+When(/^I follow the left menu "([^"]*)"$/) do |arg1|
```

Comment:
It would be nice to reduce it in only one method, parse the argument and split using ">"

Reply:
Do we have more than 3 ?

Reply:
No no, luckily we don't, and I would fight against having more :wink:

Reply:
Having the same code repeated in three methods is not clean code, it's harder to maintain.

Reply:
I totally agree @srbarrios .
This was the initial proposal to **firstly** unify the way we search for menu links and how it should behave. Then I see improvements and refactoring as `tech-debt`, but they are also on their way (don't hold your breath btw :wink )

Reply:
Factorizing to the excess is not always a good idea...

Reply:
Oki, cool, just take a look tomorrow to my suggested refactor for those three methods ;)

Reply:
@srbarrios Absolutely yes. The direct link is still there, it is just squashed with the menu element text.

The new menu behaves like this:

```html
<ul>
  <li>
    <div> <!-- click here to open/close the submenu, if any -->
      <a>Level1</a> <!-- click here to load directly the first element of the submenu, Level2-1 in this case -->
    </div>
    <ul>
      <li>
        <div>
          <a>Level2-1</a>
        </div>
      </li>
      <li>
        <div>
          <a>Level2-2</a>
        </div>
      </li>
    </ul>
  </li>
</ul>
```

This means that if the step says
`When I follow the left menu "Level1" > "Level2-1"`
or just
`When I follow the left menu "Level1"`
that's the same thing in the end, because of the new implementation where if you pass the second level argument it will open the submenu and simulate the click on the sublevel, if you don't it will just trigger the click on the first level link.

The problem is different if you want to load a different page than the first option in the submenu, let's say `Level2-2`, then you will have to give the complete path and in that case I think it is still useful to do the `open/close` visibility of the submenu.

Bottom line: for the testsuite usage, I agree it will be faster to use the direct link click. Let's do the following: right now I refactored inserting everywhere (where I did not overlooked :wink: ) the complete menu path. Later we can still remove the last argument where we want to use the direct link instead. What do you think?